poutyne/framework/callbacks/lr_scheduler.py

```diff
@@ -48,6 +40,22 @@ def save_state(self, f):
         torch.save(self.scheduler.state_dict(), f)
 
 
+class _PyTorchLRSchedulerWrapper(LRScheduler):
+    def __init__(self, torch_lr_scheduler, *args, **kwargs):
+        super(_PyTorchLRSchedulerWrapper, self).__init__(*args, **kwargs)
+        self.torch_lr_scheduler = torch_lr_scheduler
+
+    def on_train_begin(self, logs):
+        optimizer = self.model.optimizer
+        self.scheduler = self.torch_lr_scheduler(optimizer, *self.args, **self.kwargs)
+
+        # Load state if the scheduler was not initialized when the user asked
+        # to load its state
+        if self.loaded_state is not None:
+            self.scheduler.load_state_dict(self.loaded_state)
+            self.loaded_state = None
```

Comment:
I guess because we don't want to load the state again if the optimizer is reused? It also frees memory. I guess it doesn't hurt to do it.

Reply:
When we create the class `self.loaded_state = None`, I don't see when we have a valid if to enter this code. And I think it's not very clear why we put it again at None (in the code).

Reply:
See load_state method.

Reply:
The actual purpose of the loaded_state attribute does not appear clear to me neither when I try to examine it. 

From what I understand, the load_state method is used separately from the on_train_begin mainly because load_state accepts a filename argument, which is not usually so pretty to pass down using a conventionnal callback (correct me if i'm wrong @freud14 ). Since this code was already present and working and changing this would require verifying all uses of LRSchedulers to ensure the states will correctly be loaded, I suggest we currently keep the current load_state and on_train_begin separation as is.

Regarding the type of the loaded_state attribute, since its only assignation is in load_state where it is assigned the return of torch.load function and the argument passed to the load_state_dict function of the Pytorch LRScheduler, it cannot be changed to a boolean.

Reply:
After self discussion, I chose to rename loaded_state to state_to_load.